src/wmtk/EdgeMeshOperationExecutor.cpp

```
@@ -0,0 +1,238 @@
+
+    // update hash on neighborhood
+    cell_ids_to_update_hash.emplace_back(m_mesh.id_edge(m_operating_tuple));
+    if (!m_mesh.is_boundary(m_operating_tuple)) {
+        m_neighbor_eids[0] = m_mesh.id_edge(m_mesh.switch_edge(m_operating_tuple));
+        cell_ids_to_update_hash.emplace_back(m_neighbor_eids[0]);
+    }
+    if (!m_mesh.is_boundary(operating_tuple_switch_vertex)) {
+        m_neighbor_eids[1] = m_mesh.id_edge(m_mesh.switch_edge(operating_tuple_switch_vertex));
+        cell_ids_to_update_hash.emplace_back(m_neighbor_eids[1]);
+    }
+
+    if (m_neighbor_eids[0] == m_neighbor_eids[1] && m_neighbor_eids[0] == m_operating_edge_id) {
+        m_is_self_loop = true;
+    }
+}
+
+void EdgeMesh::EdgeMeshOperationExecutor::delete_simplices()
+{
+    for (size_t d = 0; d < simplex_ids_to_delete.size(); ++d) {
+        for (const long id : simplex_ids_to_delete[d]) {
+            flag_accessors[d].index_access().scalar_attribute(id) = 0;
+        }
+    }
+}
+
+void EdgeMesh::EdgeMeshOperationExecutor::update_cell_hash()
+{
+    m_mesh.update_cell_hashes(cell_ids_to_update_hash, hash_accessor);
+}
+
+const std::array<std::vector<long>, 2>
+EdgeMesh::EdgeMeshOperationExecutor::get_split_simplices_to_delete(
+    const Tuple& tuple,
+    const EdgeMesh& m)
+{
+    std::array<std::vector<long>, 2> ids;
+    ids[1].emplace_back(m.id_edge(tuple));
+    return ids;
+}
+
+const std::array<std::vector<long>, 2>
+EdgeMesh::EdgeMeshOperationExecutor::get_collapse_simplices_to_delete(
+    const Tuple& tuple,
+    const EdgeMesh& m)
+{
+    std::array<std::vector<long>, 2> ids;
+    ids[0].emplace_back(m.id_vertex(tuple));
+    ids[1].emplace_back(m.id_edge(tuple));
+    return ids;
+}
+
+std::vector<Tuple> EdgeMesh::EdgeMeshOperationExecutor::prepare_operating_tuples_for_child_meshes()
+    const
+{
+    // this function is designed as a helper for multi_mesh
+    return MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, m_operating_tuple);
```

should throw an exception if not tested